vendor/k8s.io/api/core/v1/types.go

```diff
@@ -4146,6 +4146,20 @@ const (
 	ServiceTypeExternalName ServiceType = "ExternalName"
 )
 
+// DNS Mode string describes the mode used to generate fqdn/ip for communication between nodes
+// +enum
+type DnsMode string
+
+const (
+	// DnsModeServiceMesh means a FQDN will be generated, assumming the FQDN is resolvable
+	// and available in all clusters
+	DnsModeServiceMesh DnsMode = "ServiceMesh"
+
+	// DnsModeInternal means a FQDN (svc.cluster.local), a ClusterIP or a public IP will be used
+	// depending on how the service is exposed
+	DnsModeInternal DnsMode = "Internal"
+)
+
```

Comment:
Somehow I missed this diff and thought you're using a type that exists in `core/v1` API. I don't think we should change our vendored dependencies. I'm not against this enum, let's put it to our api package (`psmdb_types.go`). Also in `psmdb_defaults.go`, we need to ensure it has a default value (`Internal`) if the user hasn't specified anything.

Reply:
I agree, I didn't even notice that this was a vendor defined type myself. I've moved the enum to psmdb_types.go and added a default as you suggested.